src/components/OrderableListV2/DropZone.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { useDrop } from "react-dnd";
+import { merge } from "@utilities/merge";
+import { OrderableListItem } from "@components/OrderableListV2/types";
+import { DropZonePosition } from "@utilities/dnd";
+
+export type OnDropCallback<T> = (
+    targetItem: OrderableListItem<T>,
+    sourceItem: OrderableListItem<T>,
+    position: DropZonePosition,
+) => void;
+
+type DropZoneData<T> = {
+    targetItem: OrderableListItem<T>;
+    position: DropZonePosition;
+};
+
+export type DropZoneProps<T> = {
+    data: DropZoneData<T>;
+    onDrop?: OnDropCallback<T>;
+    children?: JSX.Element;
```

Comment:
Because in Tree, we need to wrap the element in the dropzone to enable drag&dropping stuff on/into an element. For dropzones before/after an item, the dropzone produces a standalone element